clients/pkg/logentry/stages/drop.go

```diff
@@ -21,17 +22,20 @@ const (
 	ErrDropStageInvalidConfig   = "drop stage config error, `value` and `expression` cannot both be defined at the same time."
 	ErrDropStageInvalidRegex    = "drop stage regex compilation error: %v"
 	ErrDropStageInvalidByteSize = "drop stage failed to parse longer_than to bytes: %v"
+	ErrDropStageInvalidSource   = "drop stage source invalid type should be string or list of strings"
 )
 
 var (
 	defaultDropReason = "drop_stage"
+	defaultSeparator  = ";"
 )
 
 // DropConfig contains the configuration for a dropStage
 type DropConfig struct {
-	DropReason *string `mapstructure:"drop_counter_reason"`
-	Source     *string `mapstructure:"source"`
-	Value      *string `mapstructure:"value"`
```

Comment:
Removing this is a breaking change. We could keep it and use it for the regexp match if set. If value and regexp are defined the config validation should fail.

Reply:
It's much more complicated to maintain the code handling all the exceptions and expressing them in the documentation for the end-user. Moreover, you have to have a plan to later remove them from code and docs. So I would fail the validation if an obsolete `value` will appear in the configuration.

Reply:
> So I would fail the validation if an obsolete `value` will appear in the configuration.

If it's still possible to define the `value` it's ok. I've just looked at the [documentation](https://grafana.com/docs/loki/latest/clients/promtail/stages/drop/#drop-stage-schema). I thought `expression` was new but it wasn't. So why not keep `value`? It should still work for concatenated sources.

Reply:
I was writing about this in a first comment:
> I also noticed that currently, promtail uses options value(exact match) and expression (regexp match) with configuring the drop stage it can be limited to just expression without causing confused what should be used and simplifying code

Reply:
Sorry for missing that. It seems we missed a transition here. However, we cannot break the configuration until a major version. Unfortunately we need to keep `value` to keep the Promtail configs out there valid. I'm wondering if the `expression` could be set to `mt.Sprintf("^%$", Value)` if `value` is set but `expression` is not. This would have some performance implications, though. What do you think?

Reply:
Sure I can revert the `value` in the near future. 

> I'm wondering if the expression could be set to mt.Sprintf("^%$", Value) if value is set but expression is not

It should be also combined with [regexp.QuoteMeta](https://pkg.go.dev/regexp?utm_source=godoc#QuoteMeta).

Reply:
@witalisoft would you like me to take over?

Reply:
no, I will prepare the changes

Reply:
@jeschkies ready for the review